launch.sh

```diff
@@ -10,11 +10,37 @@ until pg_isready; do
 done
 
 echo "Postgres is up, loading chado"
-su postgres -c 'createdb apollo'
-su postgres -c 'createdb chado'
-su postgres -c 'psql -f /apollo/user.sql'
 
-su postgres -c 'PGPASSWORD=apollo psql -U apollo -h 127.0.0.1 chado -f /chado.sql'
+WEBAPOLLO_DB_HOST="${WEBAPOLLO_DB_HOST:-127.0.0.1}"
+WEBAPOLLO_DB_NAME="${WEBAPOLLO_DB_NAME:-apollo}"
+WEBAPOLLO_DB_USERNAME="${WEBAPOLLO_DB_USERNAME:-apollo}"
+WEBAPOLLO_DB_PASSWORD="${WEBAPOLLO_DB_PASSWORD:-apollo}"
+WEBAPOLLO_HOST_FLAG="-h ${WEBAPOLLO_DB_HOST}"
+
+CHADO_DB_HOST="${CHADO_DB_HOST:-127.0.0.1}"
+CHADO_DB_NAME="${CHADO_DB_NAME:-apollo}"
+CHADO_DB_USERNAME="${CHADO_DB_USERNAME:-apollo}"
+CHADO_DB_PASSWORD="${CHADO_DB_PASSWORD:-apollo}"
+CHADO_HOST_FLAG="-h ${CHADO_DB_HOST}"
+
+
+su postgres -c "psql -lqt | cut -d \| -f 1 | grep -qw $WEBAPOLLO_DB_NAME"
+if [[ "$?" == "1" ]]; then
+	echo "Apollo database not found, creating..."
+	su postgres -c "createdb $WEBAPOLLO_HOST_FLAG $WEBAPOLLO_DB_NAME"
+	su postgres -c "psql $WEBAPOLLO_HOST_FLAG -c \"CREATE USER $WEBAPOLLO_DB_USERNAME WITH PASSWORD '$WEBAPOLLO_DB_PASSWORD';\""
+	su postgres -c "psql $WEBAPOLLO_HOST_FLAG -c 'GRANT ALL PRIVILEGES ON DATABASE \"$WEBAPOLLO_DB_NAME\" to $WEBAPOLLO_DB_USERNAME;'"
+fi
+
+su postgres -c "psql -lqt | cut -d \| -f 1 | grep -qw $CHADO_DB_NAME"
+if [[ "$?" == "1" ]]; then
+	echo "Chado database not found, creating..."
+	su postgres -c "createdb $CHADO_HOST_FLAG $CHADO_DB_NAME"
+	su postgres -c "psql $CHADO_HOST_FLAG -c \"CREATE USER $CHADO_DB_USERNAME WITH PASSWORD '$CHADO_DB_PASSWORD';\""
+	su postgres -c "psql $CHADO_HOST_FLAG -c 'GRANT ALL PRIVILEGES ON DATABASE \"$CHADO_DB_NAME\" to $CHADO_DB_USERNAME;'"
+	su postgres -c "psql -U $CHADO_DB_USERNAME -h ${CHADO_DB_HOST} $CHADO_DB_NAME -f /chado.sql"
```

Comment:
You set the username and host here, but not on any of the other commands. Is this intentional? You should be setting these variables uniformly across all psql commands.

I recommend setting the PGHOST/PGUSER and other postgres env vars just before you execute a block of postgres commands that connect to a single database.